### PR TITLE
DevStudio build options implementation cleaning

### PR DESCRIPTION
**Pull request by Arthoni**
* Adjusted variable names
* Fixed interface between main window and build status widget

### CODE REVIEW

File: src/apps/openfluid-devstudio/WareBuildStatusWidget.cpp [hunk 1]
```diff
@@ -53,7 +53,33 @@ WareBuildStatusWidget::WareBuildStatusWidget(QWidget* Parent):
   ui->buildOptionsButton->setIcon(openfluid::ui::common::getIcon("settings","/ui/common"));
 
   connect(ui->buildOptionsButton, SIGNAL(clicked()), this, SIGNAL(settingsButtonClicked()));
-  connect(ui->jobStatus, SIGNAL(scrolled(bool)), this, SIGNAL(jobsScrolled(bool)));
+  connect(ui->jobStatus, SIGNAL(scrolled(bool)), this, SLOT(onJobsScrolled(bool)));
+  connect(ui->configStatus,SIGNAL(clicked()),this,SLOT(toggleModeBuildOption()));
+  connect(ui->buildStatus,SIGNAL(clicked()),this,SLOT(toggleInstalledBuildOption()));
+  connect(ui->jobStatus,SIGNAL(clicked()),this,SLOT(onJobsClicked()));
```
Comment (jctophefabre) [inline src/apps/openfluid-devstudio/WareBuildStatusWidget.cpp:59]: keep consistency on slots names
* onJobsScrolled()
* onJobsClicked()
* onBuildModeClicked()
* onConfigureModeClicked()

File: src/apps/openfluid-devstudio/WareBuildStatusWidget.cpp [hunk 2]
```diff
@@ -71,27 +97,109 @@ WareBuildStatusWidget::~WareBuildStatusWidget()
...
+
+
+// =====================================================================
+// =====================================================================
+
+
+void WareBuildStatusWidget::onJobsClicked()
+{
+  m_BuildOptions.IsMultipleJobs = !m_BuildOptions.IsMultipleJobs;
+  emit settingsChanged(m_BuildOptions);
```
Comment (jctophefabre) [inline src/apps/openfluid-devstudio/WareBuildStatusWidget.cpp:156]: as Qt signals are blocking, emit them at the end, as the last instruction of task

File: src/apps/openfluid-devstudio/WareBuildStatusWidget.cpp [hunk 3]
```diff
@@ -71,27 +97,109 @@ WareBuildStatusWidget::~WareBuildStatusWidget()
...
+  {
+    if (Up)
+    {
+      m_BuildOptions.JobsNumber++;
+    }
+    else if (m_BuildOptions.JobsNumber > 2)
+    {
+      m_BuildOptions.JobsNumber--;
+    }
+    emit settingsChanged(m_BuildOptions);
```
Comment (jctophefabre) [inline src/apps/openfluid-devstudio/WareBuildStatusWidget.cpp:177]: as Qt signals are blocking, emit them at the end, as the last instruction of task

File: src/apps/openfluid-devstudio/WareBuildStatusWidget.cpp [hunk 4]
```diff
@@ -71,27 +97,109 @@ WareBuildStatusWidget::~WareBuildStatusWidget()
...
+
+// =====================================================================
+// =====================================================================
+
+
+void WareBuildStatusWidget::toggleModeBuildOption()
+{
+
+  m_BuildOptions.IsReleaseMode = !m_BuildOptions.IsReleaseMode;
+  emit settingsChanged((m_BuildOptions));
```
Comment (jctophefabre) [inline src/apps/openfluid-devstudio/WareBuildStatusWidget.cpp:191]: as Qt signals are blocking, emit them at the end, as the last instruction of task

File: src/apps/openfluid-devstudio/WareBuildStatusWidget.cpp [hunk 5]
```diff
@@ -71,27 +97,109 @@ WareBuildStatusWidget::~WareBuildStatusWidget()
...
 // =====================================================================
 // =====================================================================
 
 
-openfluid::ui::common::ActionLabel* WareBuildStatusWidget::jobsLabel()
+void WareBuildStatusWidget::toggleInstalledBuildOption()
 {
-  return ui->jobStatus;
+  m_BuildOptions.IsWithInstall = !m_BuildOptions.IsWithInstall;
+  emit settingsChanged(m_BuildOptions);
```
Comment (jctophefabre) [inline src/apps/openfluid-devstudio/WareBuildStatusWidget.cpp:203]: as Qt signals are blocking, emit them at the end, as the last instruction of task

File: src/apps/openfluid-devstudio/WareBuildStatusWidget.hpp [hunk 6]
```diff
@@ -74,12 +95,7 @@ class OPENFLUID_API WareBuildStatusWidget : public QWidget
 
     ~WareBuildStatusWidget();
 
-    openfluid::ui::common::ActionLabel* installLabel();
-
-    openfluid::ui::common::ActionLabel* configureModeLabel();
-
-    openfluid::ui::common::ActionLabel* jobsLabel();
-
+    void injectBuildOptions(openfluid::waresdev::WareBuildOptions BuildOptions);
```
Comment (jctophefabre) [inline src/apps/openfluid-devstudio/WareBuildStatusWidget.hpp:98]: use "set" instead of "inject"

File: CMake.in.cmake [hunk 7]
```diff
@@ -23,7 +23,7 @@ SET(OFBUILD_CUSTOM_CMAKE_VERSION "${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}.
 SET(OPENFLUID_VERSION_MAJOR 2)
 SET(OPENFLUID_VERSION_MINOR 2)
 SET(OPENFLUID_VERSION_PATCH 0)
-SET(OPENFLUID_VERSION_STATUS "alpha27") # example: SET(OPENFLUID_VERSION_STATUS "rc1")
+SET(OPENFLUID_VERSION_STATUS "alpha28") # example: SET(OPENFLUID_VERSION_STATUS "rc1")
```
Comment (jctophefabre) [inline CMake.in.cmake:26]: incrementing the alpha version is not necessary

File: src/apps/openfluid-devstudio/MainWindow.cpp [hunk 8]
```diff
@@ -337,72 +329,16 @@ MainWindow::~MainWindow()
...
-
-// =====================================================================
-// =====================================================================
-
-
 void MainWindow::displayBuildOptionsDialog()
 {
-  openfluid::ui::waresdev::WareBuildOptionsDialog SettingsDialog(m_WareBuildOptions, QApplication::activeWindow());
+  openfluid::waresdev::WareBuildOptions WareBuildOptions;
+  WareBuildOptions.setDefaultsFromWorkspaceManager();
```
Comment (jctophefabre) [inline src/apps/openfluid-devstudio/MainWindow.cpp:335]: This should be included in the default beahviour of the WareBuildOptions constructor

File: src/apps/openfluid-devstudio/MainWindow.cpp [hunk 9]
```diff
@@ -744,14 +603,50 @@ void MainWindow::onExportWareSourcesAsked()
 // =====================================================================
 
 
+void MainWindow::applyBuildOptionsToWidgetCollection()
```
Comment (jctophefabre) [inline src/apps/openfluid-devstudio/MainWindow.cpp:606]: In a correct workflow, we should not update each item of the Widget collection with the new build options, but each widget should use common build options stored in only one place. Can be held by the WorkspaceManager?
This modification is maybe too deep to be implemented in this PR?

File: src/apps/openfluid-devstudio/WareBuildStatusWidget.hpp (hunk 6 above)
Comment (Arthoni) [reply src/apps/openfluid-devstudio/WareBuildStatusWidget.hpp:98]: I assumed "set" was reserved for pure setter functions. Here it is set + refresh visual, still applies ?

**Pull request merged**